lib/spack/docs/packaging_guide.rst

```diff
@@ -3453,6 +3442,12 @@ directory, including cached virtual indices.
 
 To remove all of the above, the command can be called with ``--all``.
 
+Also, cleans up temporary files for a particular package, by deleting the
+expanded/checked out source code *and* any downloaded archive.  If
+``fetch``, ``stage``, or ``install`` are run again after this, Spack's
+build process will start from scratch.
```

Comment:
I'm not sure if that's true:
```
$ spack clean zlib
$ spack fetch zlib
==> Using cached archive: /Users/Adam/spack/var/spack/cache/zlib/zlib-1.2.11.tar.gz
```
`spack clean foo` doesn't delete the archived tarball. Or are we talking about different tarballs? Maybe you're talking about the one copied to the stage? I was talking about the one in the downloads mirror.

Anyway, I would like to see this behavior:
```
$ spack clean --stage foo  # delete stage for foo
$ spack clean foo  # same as above
$ spack clean --downloads foo  # delete downloaded archive for foo
$ spack clean --stage --downloads foo  # both of the above
```
As long as I can do this, I'm happy. Not sure how much work would be involved in `spack clean --downloads foo` though.